pkg/disttask/framework/scheduler/interface.go

```diff
@@ -59,7 +59,7 @@ type TaskManager interface {
 	// we only consider pending/running subtasks, subtasks related to revert are
 	// not considered.
 	GetUsedSlotsOnNodes(ctx context.Context) (map[string]int, error)
-	GetSubtaskInStatesCnt(ctx context.Context, taskID int64, states ...proto.SubtaskState) (int64, error)
+	GetSubtaskCntByStates(ctx context.Context, taskID int64, step proto.Step) (map[proto.SubtaskState]int64, error)
```

Comment:
```suggestion
	GetSubtaskCntByStep(ctx context.Context, taskID int64, step proto.Step) (map[proto.SubtaskState]int64, error)
```

Reply:
the `ByStates` means group by state. will change to InStates to avoid conflict with normal `ByXXX`

Reply:
Indeed exist confusion

> conflict with normal ByXXX

Reply:
GetSubtaskCntGroupByStates?

Reply:
https://github.com/pingcap/tidb/pull/49971/commits/6d9a55aa98ce7582a2fe91bb3065127b7d6e1ab2